src/Language/SystemVerilog/Parser/Preprocess.hs

```diff
@@ -148,6 +158,64 @@ isIdentChar ch =
     ('0' <= ch && ch <= '9') ||
     (ch == '_') || (ch == '$')
 
+-- Update a Cond for an `else block: if we've never had a hit, this is the one
+-- for us. Otherwise a previous block was true, so it isn't.
+elseCond :: Cond -> Cond
+elseCond NeverTrue = CurrentlyTrue
+elseCond _ = NeverTrue
+
+-- Return a Cond for whether a name is defined (assuming that it hasn't been
+-- defined previously).
+ifdefCond :: Env -> String -> Cond
+ifdefCond env name = if Map.member name env then CurrentlyTrue else NeverTrue
```

Comment:
Yes, I think you're right (and laziness means I don't believe there's any change in performance). I've implemented `boolToCond` instead and am threading bools through rather than the map and string.